Sb cross receiver samples (#39514)
Azure/azure-sdk-for-net: sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample16_CrossReceiverMessageSettlement.md

# Cross Receiver Message Settlement

The message settlement APIs on the `ServiceBusReceiver` require passing in the
`ServiceBusReceivedMessage`. This can be limiting for scenarios in which the message needs to be persisted and then
settled across process boundaries. There are two strategies that can be used for settling a message with a different

> receiver. The correct strategy to use will depend on the specific scenario.

Reviewer: ```suggestion
receiver. The recommended strategy depends on the specific application scenario.
```
My intent here was to preserve what you were saying but soften the tone just a bit.

> ## Storing the Entire `ServiceBusReceivedMessage`

Reviewer: ```suggestion
## Storing the entire `ServiceBusReceivedMessage`
```

```C# Snippet:ServiceBusWriteReceivedMessage
var client1 = new ServiceBusClient(connectionString);
ServiceBusSender sender = client1.CreateSender(queueName);

var message = new ServiceBusMessage("some message");
await sender.SendMessageAsync(message);

ServiceBusReceiver receiver1 = client1.CreateReceiver(queueName);
ServiceBusReceivedMessage receivedMessage = await receiver1.ReceiveMessageAsync();
ReadOnlyMemory<byte> amqpMessageBytes = receivedMessage.GetRawAmqpMessage().ToBytes().ToMemory();
ReadOnlyMemory<byte> lockTokenBytes = Guid.Parse(receivedMessage.LockToken).ToByteArray();
```

In order to rehydrate the message in another process, we would do the following:

```C# Snippet:ServiceBusReadReceivedMessage
AmqpAnnotatedMessage amqpMessage = AmqpAnnotatedMessage.FromBytes(new BinaryData(amqpMessageBytes));
ServiceBusReceivedMessage rehydratedMessage = ServiceBusReceivedMessage.FromAmqpMessage(amqpMessage, new BinaryData(lockTokenBytes));

var client2 = new ServiceBusClient(connectionString);
ServiceBusReceiver receiver2 = client2.CreateReceiver(queueName);
await receiver2.CompleteMessageAsync(rehydratedMessage);
```

> ## Storing Only the Lock Token

Reviewer: ```suggestion
## Storing only the lock token
```

> If the entire message is not needed when settling the message in a different process, you can simply store off the

Reviewer: ```suggestion
If the entire message is not needed when settling the message in a different process, you can simply preserve the
```

> lock token. In the example below, we store off the lock token with the GUID bytes. You can also simply store a

Reviewer: ```suggestion
lock token. In the example below, we store off the lock token using its GUID bytes. You can also simply store a
```

string if that is easier for your scenario.

```C# Snippet:ServiceBusWriteReceivedMessageLockToken
var client1 = new ServiceBusClient(connectionString);
ServiceBusSender sender = client1.CreateSender(queueName);

var message = new ServiceBusMessage("some message");
await sender.SendMessageAsync(message);

ServiceBusReceiver receiver1 = client1.CreateReceiver(queueName);
ServiceBusReceivedMessage receivedMessage = await receiver1.ReceiveMessageAsync();
ReadOnlyMemory<byte> lockTokenBytes = Guid.Parse(receivedMessage.LockToken).ToByteArray();
```

> 
> In order to rehydrate the message in another process using the lock token, we would do the following:  

Reviewer: ```suggestion
In order to rehydrate the message in another process using the lock token, we would do the following:  
```
